Fix Invite new users documentation

shubham-padia opened the pull request:
Fixes https://github.com/zulip/zulip/issues/30715.

For the zulip cloud instructions, for the default `Users` tab also, we will show the instruction to select that tab, since there is a possibility that the user reading the documentation is not on the User tab. This will affect any places where the `settings_tab|users` link was present. 1 screenshot has been provided in this PR for reference but the change will affect multiple pages.

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

| Send email invitations - self hosted | Send email invitations - cloud|
| -- | -- |
| <img width="891" alt="send_email_invitation_commit2_new" src="https://github.com/zulip/zulip/assets/13910561/fa997a3b-1f92-4baf-91ff-0631393fb93b"> | <img width="907" alt="send_email_invitation_cloud" src="https://github.com/zulip/zulip/assets/13910561/5d043d88-faa8-4b30-931d-1df2ec47d621">
| Reusable invitation link - self hosted | Reusable invitation link - cloud |
| <img width="895" alt="reusable_invitation_link_commit2_new" src="https://github.com/zulip/zulip/assets/13910561/750a61c5-fcd6-430d-9b50-e30ea2ed9967"> | <img width="914" alt="reusable_invitation_link_cloud" src="https://github.com/zulip/zulip/assets/13910561/9eadae18-43fd-43bf-a8e4-e45c3b6af167">
| Manage pending invitations - self hosted | Manage pending invitations - cloud |
| <img width="900" alt="manage_invitation_link_commit2_new" src="https://github.com/zulip/zulip/assets/13910561/d65b92a3-712f-4fbe-a60d-921f509f85fe"> | <img width="910" alt="manage_invitation_cloud" src="https://github.com/zulip/zulip/assets/13910561/a4d2edf4-2067-470a-9330-1ec5c3319a1a">
| Reactivate a user - self hosted | Reactivate a user - cloud |
| <img width="900" alt="Screenshot 2024-07-08 at 7 29 48 PM" src="https://github.com/zulip/zulip/assets/13910561/f78159bf-2b6d-4574-90b5-34b2968f1d25"> | <img width="899" alt="reactivate_a_user_cloud" src="https://github.com/zulip/zulip/assets/13910561/3a208b65-7029-4278-b2c6-8d9f69821e20">
| Deactivate a user - self hosted | Deactivate a user - cloud |
| <img width="903" alt="deactivate_a_user_self_hosted" src="https://github.com/zulip/zulip/assets/13910561/f6d84553-f512-4d78-ab2c-750f22eb31e3"> | <img width="912" alt="deactivate_a_user_cloud" src="https://github.com/zulip/zulip/assets/13910561/039cb1df-0940-4740-aecf-f8cca6ee9710">


**Self review checklist:**
<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

shubham-padia committed:
help: Add missing step to invite users docs.

Fixes #30715.
The docs were missing a step to click on the link with the text `Invite
users to organization`. The name of the tabs we see after clicking the
link were also out of date, so they have also been updated.
A link to the "Invitations" tab has been added replacing the hardcoded
manual instructions via the gear icon.
"Invitations" has also been added to the `Navigate to ..` style of
writing the instructions for self hosted instances.

shubham-padia committed:
help: Add missed rename of the Deactivated users tab settings link.

The deactivated tab link was shown with the text `Users` while it
should have said `Deactivated users`. We also remove 1 redundant
instruction asking to select the `Deactivated` tab.

alya commented:
Please add screenshots of both self-hosted and Zulip Cloud instructions (see [here](https://chat.zulip.org/#narrow/stream/49-development-help/topic/Cloud.20.2Fhelp.20pages.20on.20dev/near/1845661)).

alya commented:
We don't include icons for settings tabs, so the one on Users should be dropped.

alya commented:
Please follow the pattern on `/help/deactivate-or-reactivate-a-user` for logged in instructions, so the first instruction should look like this (for Invitations, rather than Users, of course):

![Screenshot 2024-07-08 at 00 13 48](https://github.com/zulip/zulip/assets/2090066/476f2b86-d66e-458e-8f29-25003e93c28d)

shubham-padia committed:
help: Add a different set of cloud instructions for Users tabs.

For the 3 Users tabs, `Invitations`, `Users` and `Deactivated`, we were
using the standard instructions for Zulip cloud users. This led to false
instructions like `On the left, click **Invitations**` when
`Invitations` was not a part of the left sidebar of organisation
settings.
We also added the default tab `Users` as part of these
instructions since it can happen that the user navigating the
documentation might be on some other tab than the default and might get
confused.

shubham-padia commented on zerver/tests/test_docs.py:
```diff
@@ -406,6 +406,26 @@ def test_help_settings_links(self) -> None:
+        # The full sentence hasn't been tested since there's a bunch of
+        # whitespace in the HTML after the invitations link and we
+        # don't want that whitespace to be part of the test since it
+        # might change in the future.
```
I'm not sure whether I should test the full sentence or not because of the whitespace. I saw that the other tests are also not testing the full sentences and decided to do something similar here. See the whitespace after invitations link in the code block below

```
Navigate to the <a href="/#organization/users/invitations">Invitations</a>                     tab of the <strong>Organization settings</strong> menu.
```

shubham-padia commented:
I've made the requested changes and updated the screenshots. I've also squashed the first two commits and the commit 1 and commit 2 screenshots have been removed since we have a decision on what is the desired behaviour

laurynmm commented on zerver/lib/markdown/help_settings_links.py:
```diff
@@ -114,16 +114,32 @@
-        if setting_name in ["Bots", "Users"]:
+        if setting_name in ["Bots", "Users", "Invitations", "Deactivated"]:
             return f"1. Navigate to the {relative_link} \
                     tab of the **{setting_type_name}** menu."
```
I think we still want all of these to say "the **Users** tab of the **Organization settings** menu." since there isn't a tab in the left side of that overlay for invitations or deactivated (users).

laurynmm commented on help/invite-new-users.md:
```diff
@@ -29,9 +29,11 @@ permission to invite users.
 
 {start_tabs}
 
-{!invite-users.md!}
```
I think we want these instructions to be directly for the **Invite users** option in gear menu. Going to the **Organization settings** menu is at least a couple of extra clicks.

laurynmm commented on help/invite-new-users.md:
```diff
@@ -29,9 +29,11 @@ permission to invite users.
+1. Click on **Invite users to organization.**
 
-1. Click **Send an email**.
+1. Select the **Send invite email** tab.
```
This is the default, so we probably don't need to include it.

laurynmm commented on zerver/lib/markdown/help_settings_links.py:
```diff
@@ -114,16 +114,32 @@
+
+1. On the left, click **Users**.
+
+1. Select the {setting_reference} tab.
```
It is nice to have this step automatically generated here and not present in the relative settings link case. It just feels like we need three variables - the setting type (org/personal), the setting tab (users) and the setting reference - for this to work with the relative link case as well.

laurynmm reviewed (commented):
@shubham-padia - Went ahead and pushed some review comments. I think figuring out how to do these particular cases for settings links is worth doing, but may take some more iteration. @drrosa and I worked on some similar things in `help_relative_links.py` which might give us some ideas here as well.

alya commented:
I opened a [CZO discussion](https://chat.zulip.org/#narrow/stream/19-documentation/topic/linking.20to.20settings.20subtabs).
